### PR TITLE
Abandon the OBS build tool and use mock instead

### PR DESCRIPTION
**Pull request by brianjmurrell**
The OBS build tool has depdency resolution limitations that make it
unable to build some packages.  In particular it only recognizes a
single verison of a given package and so is *quite* unable to handle
our multi-package-version repos.

Since we are using mock for all RPM builds, we might as well just use
Fedora as the build platform to get much more recent versions of mock.

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -39,6 +39,7 @@
 // To use a test branch (i.e. PR) until it lands to master
 // I.e. for testing library changes
 //@Library(value="pipeline-lib@your_branch") _
+@Library(value="pipeline-lib@use-mock") _
```
Comment (JohnMalmberg) [inline Jenkinsfile]: Currently this should not depend on that PR branch.

Comment (brianjmurrell) [reply Jenkinsfile]: This line needs to be reverted once daos-stack/pipeline-lib#83 lands.  Or is there something I am not seeing?

Comment (JohnMalmberg) [reply Jenkinsfile]: I do not see any changes in the pipeline-lib@use-mock branch that this PR depends on.
This repository does not use the pipeline-lib jenkinsfile or the packageBuildingPipeline.groovy.

Comment (brianjmurrell) [reply Jenkinsfile]: Fair enough.  It needs to be removed before it can be landed in any case.

**Pull request merged**